src/python/pants/backend/terraform/lint/tffmt/tffmt.py

```diff
@@ -39,7 +40,7 @@ def register_options(cls, register):
 
 
 class TffmtRequest(FmtRequest):
-    pass
+    field_set_type = TerraformFieldSet
```

Comment:
(Not specific to this PR)

I find it somewhat noteworthy that as Pants evolves it's way of doing things, little idiosyncrasies come out of the woodwork (like this one). In https://github.com/pantsbuild/pants/pull/14182 I noticed `GofmtRequest` didn't inherit from `LintRequest`.

It seems to me potentially hazardous that code can "seemingly work" in one dimension or another, but then a change brings to light that it was incorrectly configured. I wonder when third-party plugins become more standard how we can be proactive about avoiding these possible idiosyncrasies.

Reply:
This particular change is because `mypy` can't check some usages of `ClassVar`, unfortunately: this is an abstract `ClassVar` that it failed to check was implemented for this subclass of `FmtRequest`.

> In #14182 I noticed `GofmtRequest` didn't inherit from `LintRequest`.

This is because `@union` doesn't actually require that a type extend any particular interface, although @kaos has prototyped changing that: #12577. There is rough consensus that we should change unions, but not exactly how.